schema.hh

```diff
@@ -809,6 +815,8 @@ public:
         return _raw._caching_options;
     }
 
+    dht::i_partitioner& get_partitioner() const;
+
```

Comment:
How does this interact with schema digests? Should we hash it into the digest (but not do that if the default partitioner is used, to avoid disturbing the current digests)?

If we every allow altering the partitioner, we must, but it seems unlikely that we ever can.

@tgrabiec opinion please

Reply:
I will rename this PR. Current description is not clear enough. The PR does not enable per table partitioner end-to-end. Just the internals of the single node are covered. I still have to deal with:
1. Making sure a table has the same partitioner on each node
2. Allowing user to set up a table-specific partitioner on table
3. Signal driver about what partitioner is used by a given table
4. Persist partitioner info for each table that does not use default partitioner.

Reply:
For (1), it depends on the use case. For example if cdc logs have their own partitioner, then all nodes will make the same decision when they create the log table and there is no need for extra communication.

(3) is very important, and also unfortunately quite difficult. At least everything still works even without it.

Reply:
I agree. For CDC everything should just work because partitioner will be fixed. The driver will be a bit less efficient because token_aware_policy probably won't work but we will fix it incrementally.

Reply:
@avikivity I have a weak preference for including the partitioner name in the digest, even if we don't allow changing it. Digest should be consistent with equality and schemas with different partitioners are not equal.

Reply:
Yes. But how do we deal with preexisting digests? They don't include the partitioner name. If it was just the name, then we could special-case the murmur3 partitioner not to be included, but there are also parameters, which vary from table to table.

Reply:
For mixed clusters, until the whole cluster is upgraded to the version with per-table partitioners we can ignore the field when digesting.

Reply:
Won't there be a lot of log noise because not all nodes switch to the new hash at the same instant?

Reply:
There could be schema pulls as a result, yes

Reply:
ok. @haaawk please do so, and test upgrades under a light load to see that it works and that log spam is not crazy.

Reply:
Will do. I'm just wondering why you hate me so much :D

Reply:
@haaawk was this done?

Reply:
Yes. partitioner name is included into digest now but only once the whole cluster is migrated to the new version that supports per table partitioners.